fix(starknetid): fix issue where starknet ID does not resolve

metalboyrick opened the pull request:
# Task name here

Fixed issue where starknet ID does not resolve. Note that this was tested on mainnet. 

## Types of change

- [ ] Feature
- [X] Bug
- [ ] Enhancement

## Comments (optional)

![image](https://github.com/user-attachments/assets/0edc10e0-d92f-4cce-b19a-ba60a4fde496)

**NEEDS YARN INSTALL** to import starknet ID sdk. Starknet React does not work for some reason and I've decided to work around using starknet id SDK, as well as to introduce conditional functioning to the hook.

Nadai2010 commented:
Perfect, I will test it with a Sepolia account.

Thank you for your support

metalboyrick commented:
@Nadai2010 do you know how to get a name in sepolia?

Nadai2010 commented:
@metalboyrick Yes, you can go to sepolia.starknet.id and create one, if this trick still works.

https://sepolia.app.starknet.id/

Nadai2010 commented:
Hi @metalboyrick, I'm testing directly from your branch, and Starknet.Id doesn't load and the address is not defined (in Sepolia).

![imagen](https://github.com/user-attachments/assets/50b5ec63-f114-4153-b06e-cfdacbc0546a)

metalboyrick commented:
communicating with starknet.id team for assistance

0xquantum3labs commented:
👀  @metalboyrick

metalboyrick commented:
@Nadai2010 @0xquantum3labs sepolia should work now 
![image](https://github.com/user-attachments/assets/382f423d-aa25-45de-8700-d4908cfcf2e1)

Nadai2010 commented:
Hi, I see some strange functions... in devnet I get the Connected Adres as undefined...undefined

![imagen](https://github.com/user-attachments/assets/22d11abd-2f99-43cb-9fcd-5a9a148c1159)

In Sepolia the Starknet ID profile works fine, but it gives me several errors, also the pre-deployed tokens of ETH and STRK give me some errors (no contract address, no found contracts...)

![image](https://github.com/user-attachments/assets/d38eeeec-e180-4ac0-8674-dd07f27f75e2)

Some of the commented errors:

![imagen](https://github.com/user-attachments/assets/99aa5baa-2581-40bc-8222-b9c12116c1d6)
![imagen](https://github.com/user-attachments/assets/0a54ead5-f76b-4bcb-a164-7f69918f4a59)

metalboyrick commented:
@Nadai2010 @0xquantum3labs @jrcarlos2000 fixed, but with the catch of not being able to fetch pfp yet

![Uploading image.png…]()

metalboyrick commented:
UPDATE: if you pulled this PR and tested it before, kindly do another `yarn install` since i removed starknet id SDK. I changed the approach to use their REST APIs instead since it does not involve direct contract calls from the front end

metalboyrick commented:
Parsing PFP might be difficult, i think not a pressing issue (or we can make it into an OD issue): 

see `user_data` field:

```json
{
    "id": "0x00000000000000000000000000000000000000000000000000000071f3be1792",
    "owner": "0x02d0e2c634d55149e62a092fc56c1ccb4e9604d6ebd716728c94a4f92c6e0e05",
    "main": true,
    "creation_date": 1732208508,
    "domain": {
        "domain": "zmsrick.stark",
        "migrated": true,
        "root": true,
        "creation_date": 1732208508,
        "expiry": 1826816508,
        "resolver": null,
        "legacy_address": null,
        "rev_address": null
    },
    "user_data": [
        {
            "field": "0x00000000000000000000000000000000000000000065766d2d61646472657373",
            "data": "0x0000000000000000000000004a3f30f65c43225fea4b11a55aec82581c13f22f"
        }
    ],
    "verifier_data": [
        {
            "verifier": "0x00000154bc2e1af9260b9e66af0e9c46fc757ff893b3ff6a85718a810baf1474",
            "field": "0x000000000000000000000000000000000000000000000000000000006e616d65",
            "data": "0x0146abdcf5045c822c9e22a1df0a960a459ec91b2ab18ab030b604f64751ef15"
        }
    ],
    "extended_verifier_data": []
}
```

jrcarlos2000 commented on packages/nextjs/app/page.tsx:
```diff
@@ -11,6 +12,7 @@ const Home = () => {
           <span className="block text-4xl font-bold">Scaffold-Stark 2</span>
         </h1>
         <ConnectedAddress />
+        <Address address="0x074a19c75acc3e5947f8515c025bc4286b56572c763acec95198fc87d1df9895"></Address>
```
Why do we have this hardcoded value here ? @metalboyrick

Nadai2010 replied:
This is my address to test the PFP, this page.tsx was uploaded unintentionally, it should be reset.

metalboyrick replied:
was debugging, forgot to remove

jrcarlos2000 commented on packages/nextjs/contracts/deployedContracts.ts:
this file cant change

metalboyrick replied:
resolved

jrcarlos2000 commented on packages/nextjs/scaffold.config.ts:
```diff
@@ -11,7 +11,7 @@ export type ScaffoldConfig = {
 
 const scaffoldConfig = {
-  targetNetworks: [chains.devnet],
+  targetNetworks: [chains.sepolia],
```
stays devnet as default

metalboyrick replied:
whoops

jrcarlos2000 commented on packages/nextjs/hooks/scaffold-stark/useScaffoldStarkProfile.tsx:
```diff
@@ -0,0 +1,84 @@
+      })
+      .catch((e) => {
+        console.dir(e);
+        console.error(`[useScaffoldStarkProfile] ` + e);
```
if i am connected with `jrcarlos2000.stark`, and i switch into `0x1` which doesnt have a profile, this handling will not overrite `jrcarlos2000.stark` , maybe we should set the profile to `undefined` so that `<Address/>` component uses the hex address `0x1`, because when calling `useAccount()`, the connected address is 0x1

metalboyrick replied:
right, should've set it to undefined here

metalboyrick replied:
fixed with latest commit

jrcarlos2000 commented on packages/nextjs/app/page.tsx:
```diff
@@ -1,6 +1,7 @@
 import Link from "next/link";
 import Image from "next/image";
 import { ConnectedAddress } from "~~/components/ConnectedAddress";
+import { Address } from "~~/components/scaffold-stark";
```
We dont use this
